### PR TITLE
Integration: add proxy test for all platforms

### PR DESCRIPTION
**Pull request by jsliacan**
User mode on all platforms. System mode on linux is used for the e2e proxy test (dedicated to linux). So here we complement that.

### CODE REVIEW

File: test/extended/util/proxy.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+	err := setCA()
+	if err != nil {
+		log.Fatalf("error setting up the CA: %s", err)
+	}
+
+	proxy := goproxy.NewProxyHttpServer()
+	proxy.OnRequest(goproxy.ReqHostMatches(regexp.MustCompile("^.*$"))).HandleConnect(goproxy.AlwaysMitm)
+
+	f, err := os.OpenFile("goproxylogfile.log", os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
```
Comment (anjannath) [inline test/extended/util/proxy.go]: might be useful to create this file inside `test/integration/out/` dir, the `out` dir is uploaded as artifact from the openshift CI tests

Comment (jsliacan) [reply test/extended/util/proxy.go]: Good point, will definitely do that. Thanks!

**Comment by anjannath** [comment]
tests are passing locally:
```
⠀ ➜  crc git:(74b64c8a) make BUNDLE_PATH=/home/anjan/Downloads/bundles/crc_libvirt_4.13.0_amd64.crcbundle PULL_SECRET_PATH=/home/anjan/Downloads/pull-secret GINKGO_OPTS="--ginkgo.label-filter \"goproxy\"" integration
=== RUN   TestTest
Running Suite: Test Suite - /home/anjan/Work/github.com/code-ready/crc/test/integration
=======================================================================================
Random Seed: 1685519925

Will run 6 of 36 specs
•••••[sudo] password for anjan:
•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 6 of 36 Specs in 1300.045 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 30 Skipped
--- PASS: TestTest (1300.05s)
PASS
ok      github.com/crc-org/crc/test/integration 1300.080s
````

File: test/extended/util/proxy.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+	err := setCA()
+	if err != nil {
+		log.Fatalf("error setting up the CA: %s", err)
+	}
+
+	proxy := goproxy.NewProxyHttpServer()
+	proxy.OnRequest(goproxy.ReqHostMatches(regexp.MustCompile("^.*$"))).HandleConnect(goproxy.AlwaysMitm)
+
+	logfile := filepath.Join("test", "integration", "out", "goproxylogfile.log")
```
Comment (gbraad) [inline test/extended/util/proxy.go:58]: 👍

**Review by gbraad — APPROVED** [review]

**Comment by gbraad** [comment]
lets have this merged soon

**Review by anjannath — APPROVED** [review]

**Pull request merged**